proto/viam/provisioning/provisioning.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package viam.provisioning.v1;
+
+option go_package = "go.viam.com/api/provisioning/v1";
+
+// ProvisioningService is used by viam-agent to retrieve device and smart machine info.
+service ProvisioningService {
+  // GetDeviceStatus is for retrieving the status of the device including networking.
+  rpc GetDeviceStatus(GetDeviceStatusRequest) returns (GetDeviceStatusResponse);
```

Comment:
We should decide if this is a "Device" or a "SmartMachine" and use the same term in the RPCs here.